src/main/java/org/sonarlint/intellij/config/global/SonarLintGlobalSettings.java

```diff
@@ -47,13 +48,52 @@
   private boolean autoTrigger = true;
   private List<SonarQubeServer> servers = new LinkedList<>();
   private List<String> fileExclusions = new LinkedList<>();
-  private Set<String> includedRules = new HashSet<>();
-  private Set<String> excludedRules = new HashSet<>();
+  @Deprecated
+  private final Set<String> includedRules = new HashSet<>();
+  @Deprecated
+  private final Set<String> excludedRules = new HashSet<>();
+  private final Map<String, Rule> rules = new HashMap<>();
 
   public static SonarLintGlobalSettings getInstance() {
     return ApplicationManager.getApplication().getComponent(SonarLintGlobalSettings.class);
   }
 
+  public void setRuleParam(String ruleKey, String paramName, String paramValue) {
+    if (!rules.containsKey(ruleKey)) {
+      rules.put(ruleKey, new Rule(true));
+    }
+    rules.get(ruleKey).params.put(paramName, paramValue);
+  }
+
+  public String getRuleParam(String ruleKey, String paramName) {
```

Comment:
Maybe naming the method `getRuleParamValue` or even `getRuleParamUserValue` would be clearer.

Reply:
I would return Optional<String>, or at least annotate `@CheckForNull`